content/gocourse-2019-05-melbourne.slide

```diff
@@ -801,6 +801,46 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
 		suite.Run(t, &scalerSuite{scaler: &rect{1, 1}})
 	}
 
+* Go for Java devs
+
+- Functions are lambdas ([[https://goplay.space/#5gm9iC7sNct][Function usage]])
+- Typed constants are enums ([[https://goplay.space/#KFO6kxGCLV0][ErrorEnum]])
+- exported == public
+- unexported == protected, not private
+- There are no classes, though most types behave like them
```

Comment:
Types do behave like classes though. You can give them methods, have them implement interfaces, and even embed them in other structs. The only difference is a `struct` usually has member variables, which makes no difference outside the package if they are unexported.

Reply:
I'm considering removing this line, since its not clear how to concisely sum up the nuances between structs in go and classes in java.
.
.
.
or `struct > class`?